### PR TITLE
Update Map Syntax

### PR DESCRIPTION
**Pull request by fubuloubu**
Closes: #1969

### Description for the changelog
Changed map syntax from `map(...)` to `HashMap[...]`

### Cute Animal Picture
weeeeee

### CODE REVIEW

File: vyper/context/types/indexable/mapping.py [hunk 1]
```diff
@@ -28,29 +27,32 @@ def get_index_type(self, node):
...
         location: DataLocation = DataLocation.UNSET,
         is_constant: bool = False,
         is_public: bool = False,
     ) -> MappingDefinition:
-        if isinstance(node, vy_ast.Name):
-            raise CompilerPanic("Node must be a call")
+        if not isinstance(node, vy_ast.Subscript):
+            raise CompilerPanic("Node must be a subscript")
+        if not isinstance(node.slice.value, vy_ast.Tuple):
+            raise CompilerPanic("Mapping must be a subscript with a Tuple Index")
```
Comment (iamdefinitelyahuman) [inline vyper/context/types/indexable/mapping.py:44]: This and the above shouldn't be `CompilerPanic` - they're easily reachable by the user.  I think `StructureException` is the most fitting.

Comment (iamdefinitelyahuman) [reply vyper/context/types/indexable/mapping.py:44]: Also, reference to `Tuple` isn't clear imo.  The average user has no idea that the 2 numbers are represented as a tuple in the AST. It should just be something like "subscript with 2 values"

Comment (fubuloubu) [reply vyper/context/types/indexable/mapping.py]: See if you like how I solved this in a887410

Comment (iamdefinitelyahuman) [reply vyper/context/types/indexable/mapping.py]: It's good!

**Pull request merged**